dm/config/task.go

```diff
@@ -472,10 +477,15 @@ func (c *TaskConfig) adjust() error {
 			if !ok {
 				return terror.ErrConfigMydumperCfgNotFound.Generate(i, inst.MydumperConfigName)
 			}
+			gobalConfigReferCount[prefixs[3]+inst.MydumperConfigName]++
 			inst.Mydumper = new(MydumperConfig)
 			*inst.Mydumper = *rule // ref mydumper config
 		}
 		if inst.Mydumper == nil {
+			if len(c.Mydumpers) != 0 && inst.MydumperThread == 0 {
+				log.L().Warn("mysql instance don't refer mydumper's configuration with mydumper-config-name, the default configuration will be used", zap.Int("mysql instance", i))
+				return terror.ErrConfigUnreferingCofig.Generate()
```

Comment:
If we let DM fallback to default config, we should not return here

and I think check `inst.MydumperThread == 0` is not necessary